Motoko team suggestions (#5062)
dfinity/motoko: doc/md/fundamentals/2-types/10-options-results.md

`username` is an optional [`Text`](https://internetcomputer.org/docs/motoko/base/Text) value that starts as `null` (no username set).

:::info [Null semantics]

> `null` is only valid as a value of an optional type (`?T`). This makes it closer to `None` in Rust, Scala, or Haskell — where the compiler enforces explicit handling of missing values.

Reviewer: ```suggestion
`null` is only valid as a value of an optional type (`?T`). This makes it similar to `None` in languages like Rust, Scala, or Haskell where the compiler enforces explicit handling of missing values.
```

let result2 = safeDivide(10, 0); // null
```

> To safely extract values from options, use the `let ... else` pattern. This is often preferred and encoraged over nested `switch` expressions when a fallback is needed.

Reviewer: ```suggestion

### Let / else

To safely extract values from options, use the `let ... else` pattern. This is often preferred and encoraged over nested `switch` expressions when a fallback is needed, as it improves readability and ensures fallbacks are handled.
```

For example, here’s a simple implementation of an option helper:

```motoko no-repl
func get<T>(option : ?T, defaultValue : T) : T {
  let ?value = option else return defaultValue;
  return value;
};

assert get(?"A", "B") == "A";
assert get(null, "B") == "B";
```

> This idiom improves readability and ensures fallback logic is always handled.

Reviewer: ```suggestion

```
